test/test.cpp

```diff
@@ -57,9 +57,9 @@ TEST_CASE("SubProductMatrix", "[linalg]") {
                   1,   -1, -2,
                   1,    1,  2;
   e << 2.5, -1, 2;
-  A.setConstant(999);
+  A.setOnes();
 
-  SubProductMatrix(e, A);
+  SubProductMatrix(e, A, A);
```

Comment:
Mmm so basically would you like me to make a test to see when A and B are different, we get the same result? I think the function should work as proposed. I'll make a test to see.